scripts/saml_script/saml_get_token_okta_idp.py

```diff
@@ -0,0 +1,254 @@
+
+# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE
+
+import re
+from urllib.parse import urlparse
+import requests
+import json
+from urllib.parse import urlparse
+import base64
+from dotenv import load_dotenv
+from bs4 import BeautifulSoup
+import sys
+import getpass
+import requests
+from configparser import ConfigParser
+import base64
+import xml.etree.ElementTree as ET
+from bs4 import BeautifulSoup
+from os.path import expanduser
+# from urlparse import urlparse, urlunparse
+from requests_ntlm import HttpNtlmAuth
+import logging 
+import win32pipe
+import os
+
+logger=logging.getLogger() 
+logger.setLevel(logging.DEBUG)
+
+# options
+OKTA_LOGIN_ENDPOINT = 'okta-login-endpoint'
+OKTA_SAML_ENDPOINT = 'okta-saml-endpoint'
+
+# receving the smal and okta url's from config json
+if len (sys.argv) >=3:
+ DEFAULT_LOGIN_ENDPOINT = sys.argv[1]
+ DEFAULT_SAML_ENDPOINT = sys.argv[2]
+ DI_URL = sys.argv[3]
+ 
+
+USERNAME = 'IDP_USERNAME'
+PASSWORD = 'IDP_PASSWORD'
+
+class OktaSamlResponse:
+
+    def __init__(self, options: dict = None):
+        self.options = options or {}
+        self.saml_response = None
+
+        self.login_endpoint = self.options.get(
+            OKTA_LOGIN_ENDPOINT) or DEFAULT_LOGIN_ENDPOINT
+        self.saml_endpoint = self.options.get(
+            OKTA_SAML_ENDPOINT) or DEFAULT_SAML_ENDPOINT
+
+        load_dotenv()
+
+    def value(self):
+        if self.saml_response is None:
+           self.__get_new()
+
+        return self.saml_response
+
+    def __get_new(self):
+        
+        session = requests.Session()
+     
+        try:
+            response = requests.get(DEFAULT_LOGIN_ENDPOINT)
+            response.raise_for_status() 
+             # Check for HTTP errors
+            # Process the response if successful
+        except requests.exceptions.RequestException as e:
+            print(f"invalid url: {e}")
+            sys.exit(0)
+        # get the sign-in page
+        sign_in_page = session.get(self.login_endpoint, verify=None)
+       # print()
+       # print("Enter the credentials for the IdP :")
+       # print()
+       # username = input("Username: ")
+        #print()
+        #password = getpass.getpass(prompt='Password: ')
+        username = "cirrus-dev@premiertest.com"
```

Comment:
Should you make this a more generic username? 
Could you add a comment telling the user that this is where they need to update the username and password?

Also please delete the commented out code above.